sktime/datatypes/_adapter/polars.py

```diff
@@ -119,19 +151,67 @@ def check_polars_frame(obj, return_metadata=False, var_name="obj", lazy=False):
         return ret(False, msg, None, return_metadata)
 
     # we now know obj is a polars DataFrame or LazyFrame
+    index_cols = []
+
+    if scitype in ["Series", "Panel", "Hierarchical"]:
+        index_cols = get_mi_cols(obj)
+        n_vars = len(index_cols)
+        scitypes_index = {
+            "Series": n_vars == 1 or n_vars == 0,
+            "Panel": n_vars == 2,
+            "Hierarchical": n_vars >= 3,
+        }
+
+        if not scitypes_index[scitype]:
+            cols_msg = (
+                f"{var_name} must have correct number of index columns for scitype, "
+                f"Series: 0 or 1, Panel: 2, Hierarchical: >= 3,"
+                f"found {len(index_cols)}, namely: {index_cols}"
+            )
+            return ret(False, cols_msg, None, return_metadata)
+
+        # check if index columns are monotonically increasing
+        if isinstance(obj, pl.DataFrame) and not is_monotonically_increasing(obj):
+            msg = (
+                f"The (time) index of {var_name} must be sorted monotonically "
+                f"increasing. Use {var_name}.sort() on columns representing "
+                f"index(__index__) to sort the index, or {var_name}.is_duplicated() "
+                f"to find duplicates."
+            )
+            return ret(False, msg, None, return_metadata)
+
+    elif scitype != "Table":
```

Comment:
better style: do this check at the start, e.g., if `scitype not in ["a", "b", "c"`